### ISSUE TITLE
Load balanced Cromwell

### ISSUE THREAD

**Issue by geoffjentry** [broadinstitute/cromwell#3339]
Modify Cromwell so as to play nicely in a load balanced environment where the individual Cromwell instances are sharing the same database

**Comment by patmagee** [comment]
@geoffjentry is the idea for this to enable horizontal scalability of cromwell?

**Comment by geoffjentry** [comment]
@patmagee yeah, albeit in an as-quick-as-possible way so it's still fairly naive. But this would allow for multiple Cromwells to share a single DB without blowing up the world.

**Comment by patmagee** [comment]
@geoffjentry that would be a nice feature. I saw that there was work done to enable a separate metadata DB. Is there any plans (or maybe there has already been some movement on this) to allow this db to be a document store/ `NoSQL` db of some sort?

**Comment by geoffjentry** [comment]
@patmagee You can do that right now if you want. Metadata is a service in the service registry which means if one wanted to write their own to handle those events however they can do so w/o even worrying about committing code to the cromwell repo (but that'd be nice). It's actually pretty easy to do for this service, if this interests you at all let me know and we can chat. As an example, check out PubSubMetadataServiceActor, which was a POC I did a while back.

That said, if one doesn't want to write their own, we do expect to provide an alternate implementation later this year, albeit it'll likely be wed to GCP tools.
